src/Carousel.php

```diff
@@ -20,244 +22,535 @@
+     *
+     * @return self A new instance with the specified tag name for the content caption placeholder.
      */
-    public function options(array $value): self
+    public function captionPlaceholderTagName(string $value): self
     {
         $new = clone $this;
-        $new->options = $value;
+        $new->captionPlaceholderTagName = $value;
 
         return $new;
     }
 
     /**
-     * Whether carousel indicators (<ol> tag with anchors to items) should be displayed or not.
+     * Sets the label for the next control button.
+     *
+     * @param string $value The label for the next control button.
+     *
+     * @return self A new instance with the specified label for the next control button.
      */
-    public function withoutShowIndicators(): self
+    public function controlNextLabel(string $value): self
     {
         $new = clone $this;
-        $new->showIndicators = false;
+        $new->controlNextLabel = $value;
 
         return $new;
     }
 
     /**
-     * Renders carousel indicators.
+     * Sets the label for the previous control button.
+     *
+     * @param string $value The label for the previous control button.
+     *
+     * @return self A new instance with the specified label for the previous control button.
      */
-    private function renderIndicators(): string
+    public function controlPrevLabel(string $value): self
     {
-        if ($this->showIndicators === false) {
-            return '';
-        }
+        $new = clone $this;
+        $new->controlPrevLabel = $value;
 
-        $indicators = [];
+        return $new;
+    }
 
-        for ($i = 0, $count = count($this->items); $i < $count; $i++) {
-            $options = ['data-bs-target' => '#' . $this->options['id'], 'data-bs-slide-to' => $i];
-            if ($i === 0) {
-                /** @psalm-suppress InvalidArgument */
-                Html::addCssClass($options, ['active' => 'active']);
-            }
-            $indicators[] = Html::tag('li', '', $options);
-        }
+    /**
+     * Set the carousel to crossfade slides instead of sliding.
+     *
+     * @param bool|null $value Whether to crossfade slides or not. Default is `true`.
+     *
+     * @return self A new instance with the specified crossfade value.
+     */
+    public function crossfade(bool|null $value = true): self
+    {
+        $new = clone $this;
+        $new->cssClass['crossfade'] = $value === true ? 'carousel-fade' : null;
 
-        $indicatorOptions = ['class' => ['carousel-indicators']];
+        return $new;
+    }
 
-        return Html::tag('ol', implode("\n", $indicators), $indicatorOptions)
-            ->encode($this->encodeTags)
-            ->render();
+    /**
+     * Disables the touch swipe functionality of the carousel.
+     *
+     * @return self A new instance with the touch swipe functionality disabled.
+     */
+    public function disableTouchSwiping(): self
```

Comment:
Should there be a boolean value as parameter?

Reply:
done.